File: data/code_pairs/cpp/1118.cpp
#include <cmath>

double DDANRM(int NEQ, const double* V, const double* WT, const double* RPAR, const int* IPAR) {
    double SUM = 0.0, VMAX = 0.0;
    for (int I = 0; I < NEQ; ++I) {
        double temp = std::abs(V[I] / WT[I]);
        if (temp > VMAX) VMAX = temp;
    }
    if (VMAX > 0.0) {
        for (int I = 0; I < NEQ; ++I) {
            SUM += std::pow((V[I] / WT[I]) / VMAX, 2);
        }
        return VMAX * std::sqrt(SUM / NEQ);
    }
    return 0.0;
}

File: data/code_pairs/fortran/1118.f90
DOUBLE PRECISION FUNCTION DDANRM (NEQ, V, WT, RPAR, IPAR)
  INTEGER  NEQ, IPAR(*)
  DOUBLE PRECISION  V(NEQ), WT(NEQ), RPAR(*)
  INTEGER  I
  DOUBLE PRECISION  SUM, VMAX
  DDANRM = 0.0D0
  VMAX = 0.0D0
  DO 10 I = 1,NEQ
    IF(ABS(V(I)/WT(I)) .GT. VMAX) VMAX = ABS(V(I)/WT(I))
10  CONTINUE
  IF(VMAX .LE. 0.0D0) GO TO 30
  SUM = 0.0D0
  DO 20 I = 1,NEQ
20  SUM = SUM + ((V(I)/WT(I))/VMAX)**2
  DDANRM = VMAX*SQRT(SUM/NEQ)
30  CONTINUE
  RETURN
END